cumulus/parachains/common/src/xcm_config.rs

```diff
@@ -78,3 +78,60 @@ impl<Location: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
 		matches!(asset.id, Concrete(ref id) if id == origin && origin == &Location::get())
 	}
 }
+
+/// Accepts an asset if it is a native asset from the system (Relay Chain or system parachain).
+pub struct ConcreteNativeAssetFromSystem;
+impl ContainsPair<MultiAsset, MultiLocation> for ConcreteNativeAssetFromSystem {
+	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+		log::trace!(target: "xcm::contains", "ConcreteNativeAssetFromSystem asset: {:?}, origin: {:?}", asset, origin);
+		let parent = MultiLocation::parent();
```

Comment:
Just thinking loud, word `Native` means `MultiLocation::parent()` here.
What about renaming `ConcreteNativeAssetFromSystem` -> `ConcreteAssetFromSystem`
and add `AssetLocation: Get<MultiLocation` to `ConcreteAssetFromSystem` like:
```
impl <AssetLocation: Get<MultiLocation`> ContainsPair<MultiAsset, MultiLocation> for ConcreteAssetFromSystem<AssetLocation>
```

We have defined native token location in all SP runtimes like:
```
pub const DotLocation: MultiLocation = MultiLocation::parent();
...
pub const KsmLocation: MultiLocation = MultiLocation::parent();
```

so I think maybe it would be a little bit clear and more flexible using it like:
```
ConcreteAssetFromSystem<DotLocation>
```

Reply:
Do you see any other use case where `AssetLocation` is going to be different to `MultiLocation::parent()` for `ConcreteAssetFromSystem`?

Reply:
well, I just dont like to define `MultiLocation::parent()` on several places,
and also it is not very clear from docs what "native asset" means in this context, but with `ConcreteAssetFromSystem<DotLocation>` it is clear without any documentation.

the only use case which comes to my mind could be the moving DOT/KSM out of the relay chain to the AssetHubs, so then it wont be `MultiLocation::parent()`

Reply:
@bkontur I changed it, check it out